std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderSheet.java

```diff
@@ -0,0 +1,163 @@
+package org.enso.table.excel.xssfreader;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.util.XMLHelper;
+import org.enso.table.excel.ExcelRow;
+import org.enso.table.excel.ExcelSheet;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+public class XSSFReaderSheet implements ExcelSheet {
+  private final int sheetIdx;
+  private final String sheetName;
+  private final String relId;
+  private final XSSFReaderWorkbook parent;
+
+  private boolean readSheetData = false;
```

Comment:
nitpick: can we make the name sound more like a flag and less like a method (verb)? e.g. `hasAlreadyBeenRead`?